python/wpt/test.py

```diff
@@ -50,48 +50,86 @@
 
 @dataclasses.dataclass
 class MockPullRequest():
+    """
+    A mock representation of a GitHub pull request.
+
+    Attributes:
+        head (str): The name of the branch where changes are made.
+        number (int): The pull request number.
+        state (str): The state of the pull request (default is "open")
+    """
     head: str
     number: int
     state: str = "open"
 
 
 class MockGitHubAPIServer():
+    """
+    A mock server to simulate GitHub API for testing purposes.
+    """
     def __init__(self, port: int):
         self.port = port
         self.disable_logging()
         self.app = flask.Flask(__name__)
         self.pulls: list[MockPullRequest] = []
 
-        class NoLoggingHandler(WSGIRequestHandler):
-            def log_message(self, *args):
-                pass
-        if logging.getLogger().level == logging.DEBUG:
-            handler = WSGIRequestHandler
-        else:
-            handler = NoLoggingHandler
-
-        self.server = make_server('localhost', self.port, self.app, handler_class=handler)
+        def __init__(self, port: int):
+        self.port = port
+        self.disable_logging()
+        self.app = flask.Flask(__name__)
+        self.pulls: List[MockPullRequest] = []
+        self.configure_routes()
```

Comment:
This doesn't even parse as valid Python? Have you tested this change by running `./mach test-scripts`?